### DISCUSSION TITLE
Configure SwaggerForOcelot running in a subpath (IIS application)

### DISCUSSION THREAD

**Discussion by BernhardJ** [Burgyn/MMLib.SwaggerForOcelot#243] [Q&A] [answered]
Hi

My ocelot api gateway is running as an application (in a subpath) of an IIS website. 
The IIS website is mapped to the path C:\web\gateway and the application is mapped to the path C:\web\gateway\ocelot.
The url of the ocelot gateway is: https://gateway.testdomain.com/ocelotpath/
See configuration IIS:
![20221031_144328](https://user-images.githubusercontent.com/3414941/199022291-970e9f90-e9ff-479d-85d5-5489b194283b.png)

I deployed the code with the following ocelot configuration:
```
{
  "Routes": [
    {
      "DownstreamPathTemplate": "/testpath/{version}/testapi",
      "DownstreamScheme": "https",
      "DownstreamHostAndPorts": [
        {
          "Host": "api.testdomain.com",
          "Port": 443
        }
      ],
      "UpstreamPathTemplate": "/{version}/testapi",
      "UpstreamHttpMethod": [ "Get" ],
      "SwaggerKey": "testapiswagger",
      "DelegatingHandlers": [
      ]
    }
  ],
  "SwaggerEndPoints": [
    {
      "Key": "testapiswagger",
      "Config": [
        {
          "Name": "Test API",
          "Version": "v1",
          "Url": "https://api.testdomain.com/testpath/swagger/v1/swagger.json"
        }
      ]
    }
  ],
  "GlobalConfiguration": {
    "BaseUrl": "https://gateway.testdomain.com/ocelotpath/"
  }
}
```
The api gateway works fine, but the swagger does not load the swagger.json.
![20221031_145124](https://user-images.githubusercontent.com/3414941/199023916-5327ff75-cde6-42e9-8953-cec21f7a164b.png)
The subpath ocelotpath is missing. If I add this manually to the url, the swagger, the json file is loaded.

If I add the subpath to the setting PathToSwaggerGenerator, then the subbath has to be twice in the path.
What is the best way to configure the swagger for ocelot in my case?

Thanks a lot

**Comment by Burgyn** [comment]
Hi @BernhardJ,

please try to follow [Virtual directory](https://github.com/Burgyn/MMLib.SwaggerForOcelot#virtual-directory) part of documentation.
Maybe that will help.

**Reply by BernhardJ** [reply]
Hi @Burgyn 

I tried this setting. This did not change anything.
I understand the virtual directory setting for the downstream service and not for the gateway itself.

**Comment by BernhardJ** [comment, marked as answer]
In the meantime I found a solution that works:

```
    app.UseSwaggerForOcelotUI(opt =>
    {
        // swaggerForOcelot options
        opt.PathToSwaggerGenerator = "/swagger/docs";
        opt.DownstreamSwaggerEndPointBasePath = "/ocelotpath/swagger/docs";
        opt.ServerOcelot = "/ocelotpath";
    });
```
Thanks for the help
